test/compare-acorn-es2020/optional-chaining.expected.js

```diff
@@ -0,0 +1,47 @@
+obj.aaa.bbb;
+obj.aaa?.bbb;
+obj?.aaa.bbb;
+obj?.aaa?.bbb;
+obj.aaa.bbb;
+obj.aaa?.bbb;
+(obj?.aaa).bbb;
+(obj?.aaa)?.bbb;
+(obj?.aaa).bbb.ccc.ddd;
+((obj?.aaa).bbb?.ccc).ddd;
+(obj?.aaa)?.bbb;
+obj[aaa][bbb];
+obj[aaa]?.[bbb];
+obj?.[aaa][bbb];
+obj?.[aaa]?.[bbb];
+obj[aaa][bbb];
+obj[aaa]?.[bbb];
+(obj?.[aaa])[bbb];
+(obj?.[aaa])?.[bbb];
+obj[aaa][bbb][ccc][ddd];
+((obj?.[aaa])[bbb]?.[ccc])[ddd];
+1?.a;
+obj()();
+obj()?.();
+obj?.()();
+obj?.()?.();
+obj()();
+obj()?.();
+(obj?.())();
+(obj?.())?.();
+obj()()()();
+((obj?.())()?.())();
+(a?.b)();
+a?.b();
+a?.b?.();
+(a?.b)?.();
+a?.().b;
+(a?.()).b;
+a?.b.c();
+(a?.b.c)();
+a.b?.().c;
+(a.b?.()).c;
+(a.b?.())?.c;
+new (a?.b().c)();
+new (a?.b())();
+new (a?.b().c)();
+new (a?.b())();
```

Comment:
The trailing parens on these four lines are unnecessary. I think we're missing some special handling in the generation of `NewExpression`.

Reply:
https://github.com/estools/escodegen/pull/412/commits/8f05fb08753c49aa1a0cc6e14c593694e3b06db8

Reply:
If you want, i can change early return with condition like:
```
        if (
                !(expr.callee.type === Syntax.ChainExpression && !length)
                && (
                    !(flags & F_ALLOW_UNPARATH_NEW)
                    || parentheses
                    || length > 0
                )
            )
```

Reply:
@michaelficarra Those parentheses were included only because the `parentheses` option [defaults to true](https://github.com/estools/escodegen/blob/eee573bee25bcc42ef405e685d6930eda62e33de/escodegen.js#L191). They were not correctly omitted in the `min.js` test.

@sanex3339 You should revert that commit.

Reply:
Ok

Reply:
https://github.com/estools/escodegen/pull/412/commits/eb9cbb0bfd1ea745c1a12632f66b31f602573ded

Reply:
Oh, GitHub collapsed the `.min.js` diff and I didn't even notice those tests. Fixed: https://github.com/estools/escodegen/commit/e8e2e81675ce74a6f538ad0274fd72ada26fb17d